controllers/iamauthpolicy_controller.go

```diff
@@ -27,17 +29,26 @@ import (
 	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
+const (
+	IAMAuthPolicyAnnotation      = "iam-auth-policy"
+	IAMAuthPolicyAnnotationResId = k8s.AnnotationPrefix + IAMAuthPolicyAnnotation + "-resource-id"
+	IAMAuthPolicyAnnotationType  = k8s.AnnotationPrefix + IAMAuthPolicyAnnotation + "-resource-type"
+	IAMAuthPolicyFinalizer       = k8s.AnnotationPrefix + IAMAuthPolicyAnnotation
```

Comment:
Is this the typical finalizer format for CRDs? We follow a different pattern in other controllers, like `service.k8s.aws/resources`

Reply:
From what I've read, finalizers are typically in the format of `example.com/finalizer`. This appears to be something we should change across all our resources, and have one finalizer for any resource type (which makes sense, since we have no reason to put the resource Kind in the finalizer for that kind, the purpose of the finalizer name is to avoid collisions with other controllers).

This is a non-blocker for the current PR, I'll make an issue to address it.

Reply:
Does not really matter, as long as each controller has own finalizer there wont be issues.